### PR TITLE
Fixes issue #5025 Text loses indentation when typing

### PR DESCRIPTION
**Pull request by mkrueger**
The virtual indent behavior was quite broken in the completion window
case. This should stop virtual indent to interfere with completion and
indent should be corrected after the completion session ends.

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.6

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionController.cs [hunk 1]
```diff
@@ -224,6 +224,8 @@ public bool ShowListWindow (ICompletionDataList list, CodeCompletionContext comp
 
 		public void HideWindow ()
 		{
+			if (CompletionWidget is ITextEditorImpl impl)
+				impl.FixVirtualIndentation ();
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionController.cs]: This call is out of place. The code completion controller shouldn't have to care about indentation.

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionController.cs]: Use case: indentation should correct when window hides.

Where should that be placed if not the control ?

Comment (slluis) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionController.cs]: Text editor deciding to fix indentation when completion window closes -> Yes
Completion window special casing editor to fix indentation when it closes -> No, not his business.

It is a matter of correctly setting responsibilities. The completion window controller controls the behavior of the completion window, not the behavior of the editor.

**Comment by slluis** [comment]
@mrward can you please verify that this PR fixes the issue?

**Comment by mrward** [comment]
Looks good to me. I cannot reproduce the problem I was seeing before 👍

**Pull request merged**